### PR TITLE
Deletion of vaccination certificates (EXPOSUREAPP-7111)

### PR DESCRIPTION
**Pull request by LukasLechnerDev**
### Description
This PR implements deletion of Vaccinations in the Vaccination List View via
- overflow menu
image

- swipe to delete
image

In case the last vaccination is deleted, the app navigates back to the HomeScreen. 

### How to test
- register some vaccinations
- try the overflow menu and swipe to delete
- See if vaccinations are properly deleted, if the last vaccination is deleted, you should be at the home screen again.

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/list/VaccinationListViewModel.kt [hunk 1]
```diff
@@ -125,6 +144,12 @@ class VaccinationListViewModel @AssistedInject constructor(
         events.postValue(Event.NavigateToVaccinationQrCodeScanScreen)
     }
 
+    fun deleteVaccination(vaccinationCertificateId: String) {
+        launch(scope = appScope) {
+            vaccinationRepository.deleteVaccinationCertificate(vaccinationCertificateId)
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/list/VaccinationListViewModel.kt:149]: delete can throw an exception ,right?  if so please catch it and report the error?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/list/VaccinationListFragment.kt [hunk 2]
```diff
@@ -82,6 +96,12 @@ class VaccinationListFragment : Fragment(R.layout.fragment_vaccination_list), Au
                             navigatorExtras
                         )
                     }
+                    is DeleteVaccinationEvent -> {
+                        showDeleteVaccinationDialog(event.vaccinationCertificateId, event.position)
+                    }
+                    is NavigateBack -> {
+                        popBackStack()
+                    }
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/list/VaccinationListFragment.kt:104]: Just a personal taste :)  , no need for `{}`  if it just one statement
```kotlin
 is DeleteVaccinationEvent -> showDeleteVaccinationDialog(event.vaccinationCertificateId, event.position)
 is NavigateBack ->  popBackStack()
```

**Review by mtwalli — COMMENTED** [review]
Looks good, few small comment/?

**Comment by mtwalli** [comment]
First two items should not be swiped, Please have a look at the CameraItem in CheckInsFragment to see how to exclude those items 
device-2021-05-18-181422

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/list/VaccinationListViewModel.kt (hunk 1 above)
Comment (LukasLechnerDev) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/list/VaccinationListViewModel.kt]: Yes, fixed.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/list/VaccinationListFragment.kt (hunk 2 above)
Comment (LukasLechnerDev) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/list/VaccinationListFragment.kt]: Adapted.

**Comment by LukasLechnerDev** [comment]
> First two items should not be swiped, Please have a look at the CameraItem in CheckInsFragment to see how to exclude those items

Should be fixed now.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/list/VaccinationListViewModel.kt [hunk 3]
```diff
@@ -125,6 +144,16 @@ class VaccinationListViewModel @AssistedInject constructor(
         events.postValue(Event.NavigateToVaccinationQrCodeScanScreen)
     }
 
+    fun deleteVaccination(vaccinationCertificateId: String) {
+        launch(scope = appScope) {
+            try {
+                vaccinationRepository.deleteVaccinationCertificate(vaccinationCertificateId)
+            } catch (exception: Exception) {
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/list/VaccinationListViewModel.kt:151]: should we show the error to the user here ? `errors.postValue(exception)`?

Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/list/VaccinationListViewModel.kt:151]: Look for `error.toErrorDialogBuilder(requireContext()).show()`

**Review by d4rken — DISMISSED** [review]
lgtm to me.
After @mtwalli's comments and UA approval, could be merged.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/list/VaccinationListViewModel.kt (hunk 3 above)
Comment (LukasLechnerDev) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/list/VaccinationListViewModel.kt:152]: Thanks, the error is now shown to the user.

**Review by NataliaLemmerth — APPROVED** [review]
UA approved

**Review by d4rken — APPROVED** [review]

**Review by mtwalli — APPROVED** [review]

**Pull request merged**